lib/cocoapods/installer/analyzer.rb

```diff
@@ -803,10 +803,15 @@ def sources
 
           # Add any sources specified using the :source flag on individual dependencies.
           dependency_sources = podfile.dependencies.map(&:podspec_repo).compact
-
           all_dependencies_have_sources = dependency_sources.count == podfile.dependencies.count
+
+          external_sources = podfile.dependencies.map(&:external_source).compact
+          all_dependencies_have_external_sources = external_sources.count == podfile.dependencies.count
```

Comment:
This is actually not correct, I would like to get also if the external source have dependencies, because in that case we should still use the default master repo. Ideas on how to do it @dnkoutso?